hazelcast/src/test/java/com/hazelcast/cache/jsr/CacheListenerTest.java

```diff
@@ -66,13 +68,21 @@
     public TestName testName = new TestName();
 
     @BeforeClass
-    public static void init() {
-        JsrTestUtil.setup();
+    public static void init() throws Exception {
+        if (ClassLoaderUtil.isClassAvailable(null, "com.hazelcast.client.HazelcastClient")) {
+            setupClientTestReflectively();
```

Comment:
Instead of duplicating all the overriding code from member to client-side `CacheListenerTest`, I opted for reflectively running the necessary client-side test initialization & tear down from the member-side setup/teardown methods.